packages/authentication/src/models/auth-user.ts

```diff
@@ -1,11 +1,48 @@
 import { generateEntityId } from "@medusajs/utils"
-import { BeforeCreate, Entity, OnInit, PrimaryKey } from "@mikro-orm/core"
+import {
+  BeforeCreate,
+  Cascade,
+  Entity,
+  ManyToOne,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import AuthProvider from "./auth-provider"
+
+type OptionalFields = "password_hash" | "app_metadata" | "user_metadata"
 
 @Entity()
 export default class AuthUser {
+  [OptionalProps]: OptionalFields
+
   @PrimaryKey({ columnType: "text" })
   id!: string
 
+  @Property({ columnType: "text", index: "IDX_auth_user_email" })
+  email: string
+
+  @Property({
+    columnType: "text",
+    nullable: true,
+    hidden: true,
+  })
+  password_hash: string | null
```

Comment:
todo: I think the password hash shouldn't be nullable, but in case a store user for example authenticate through a provider, we might generate a random password that we hash. And the user can change his password later on if needed. wdyt? I did something similar [here](https://github.com/adrien2p/medusa-plugins/blob/main/packages/medusa-plugin-auth/src/core/validate-callback.ts#L158) wdyt?

Reply:
I think it would be better to keep as nullable since it's probably just our internal auth that will use it, otherwise we should just remove it entirely and store the hash in the `app_metadata` of the user entity

Reply:
we can see later, but we will have to include guest account and real account somewhere, which for now is based on that as well